src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/MSBuildProjectLoadListener.cs

```diff
@@ -98,42 +107,24 @@ public async void ProjectLoaded(ProjectLoadedEventArgs args)
                     return;
                 }
 
-                var targetFramework = GetTargetFramework(args.ProjectInstance);
-                if (string.IsNullOrEmpty(targetFramework))
-                {
-                    // This should never be true but we're being extra careful.
-                    return;
-                }
-
+                _projectConfigurationPublisher.SetPublishFilePath(projectFilePath, configPath);
                 var razorConfiguration = GetRazorConfiguration(args.ProjectInstance, _projectConfigurationProviders);
-                var projectDirectory = args.ProjectInstance.GetPropertyValue(MSBuildProjectDirectoryPropertyName);
-                var fileSystem = RazorProjectFileSystem.Create(projectDirectory);
-                var projectEngine = _projectEngineFactory.Create(razorConfiguration, fileSystem, (builder) => { });
                 var project = _workspace.CurrentSolution.GetProject(args.Id);
-                var tagHelpers = await _tagHelperResolver.GetTagHelpersAsync(project, projectEngine, CancellationToken.None);
-                var projectWorkspaceState = new ProjectWorkspaceState(tagHelpers);
 
-                var projectConfiguration = new RazorProjectConfiguration()
+                await Task.Factory.StartNew(() =>
                 {
-                    ProjectFilePath = projectFilePath,
-                    Configuration = razorConfiguration,
-                    TargetFramework = targetFramework,
-                    ProjectWorkspaceState = projectWorkspaceState,
-                };
-
-                var serializedOutput = JsonConvert.SerializeObject(
-                    projectConfiguration,
-                    Formatting.Indented,
-                    JsonConverterCollectionExtensions.RazorConverters.ToArray());
-
-                try
-                {
-                    File.WriteAllText(configPath, serializedOutput);
-                }
-                catch (Exception)
-                {
-                    // TODO: Add retry.
-                }
+                    var projectSnapshot = _projectManager.GetLoadedProject(projectFilePath);
+                    var hostProject = new OmniSharpHostProject(projectFilePath, razorConfiguration);
+                    if (projectSnapshot == null)
+                    {
+                        _projectManager.ProjectAdded(hostProject);
```

Comment:
Now all this guy does (instead of serializing to disk) is update the project system based on MSBuild events that flow.